TypeError: unorderable types: NoneType() < str()

Hi guys! I just updated to version 1.14 via pip and got error:

```
 (master) hexlet$ make compose-bash
docker-compose run web bash
Traceback (most recent call last):
  File "/usr/local/bin/docker-compose", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.5/dist-packages/compose/cli/main.py", line 68, in main
    command()
  File "/usr/local/lib/python3.5/dist-packages/compose/cli/main.py", line 115, in perform_command
    project = project_from_options('.', options)
  File "/usr/local/lib/python3.5/dist-packages/compose/cli/command.py", line 37, in project_from_options
    override_dir=options.get('--project-directory'),
  File "/usr/local/lib/python3.5/dist-packages/compose/cli/command.py", line 91, in get_project
    config_data = config.load(config_details)
  File "/usr/local/lib/python3.5/dist-packages/compose/config/config.py", line 379, in load
    service_dicts = load_services(config_details, main_file)
  File "/usr/local/lib/python3.5/dist-packages/compose/config/config.py", line 475, in load_services
    service_config = merge_services(service_config, next_config)
  File "/usr/local/lib/python3.5/dist-packages/compose/config/config.py", line 466, in merge_services
    for name in all_service_names
  File "/usr/local/lib/python3.5/dist-packages/compose/config/config.py", line 466, in <dictcomp>
    for name in all_service_names
  File "/usr/local/lib/python3.5/dist-packages/compose/config/config.py", line 853, in merge_service_dicts_from_files
    new_service = merge_service_dicts(base, override, version)
  File "/usr/local/lib/python3.5/dist-packages/compose/config/config.py", line 929, in merge_service_dicts
    merge_ports(md, base, override)
  File "/usr/local/lib/python3.5/dist-packages/compose/config/config.py", line 962, in merge_ports
    md[field] = [item for item in sorted(merged.values())]
TypeError: unorderable types: NoneType() < str()
Makefile:72: recipe for target 'compose-bash' failed
make: *** [compose-bash] Error 1
 (master) hexlet$ cat ^C
 (master) hexlet$ docker-compose --version
docker-compose version 1.14.0, build c7bdf9e
```

any docker-compose calls throws the same error... 
```
 (feature/consul-template) hexlet$ make compose-build
docker-compose build
Traceback (most recent call last):
  File "/usr/local/bin/docker-compose", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.5/dist-packages/compose/cli/main.py", line 68, in main
    command()
  File "/usr/local/lib/python3.5/dist-packages/compose/cli/main.py", line 115, in perform_command
    project = project_from_options('.', options)
  File "/usr/local/lib/python3.5/dist-packages/compose/cli/command.py", line 37, in project_from_options
    override_dir=options.get('--project-directory'),
  File "/usr/local/lib/python3.5/dist-packages/compose/cli/command.py", line 91, in get_project
    config_data = config.load(config_details)
  File "/usr/local/lib/python3.5/dist-packages/compose/config/config.py", line 379, in load
    service_dicts = load_services(config_details, main_file)
  File "/usr/local/lib/python3.5/dist-packages/compose/config/config.py", line 475, in load_services
    service_config = merge_services(service_config, next_config)
  File "/usr/local/lib/python3.5/dist-packages/compose/config/config.py", line 466, in merge_services
    for name in all_service_names
  File "/usr/local/lib/python3.5/dist-packages/compose/config/config.py", line 466, in <dictcomp>
    for name in all_service_names
  File "/usr/local/lib/python3.5/dist-packages/compose/config/config.py", line 853, in merge_service_dicts_from_files
    new_service = merge_service_dicts(base, override, version)
  File "/usr/local/lib/python3.5/dist-packages/compose/config/config.py", line 929, in merge_service_dicts
    merge_ports(md, base, override)
  File "/usr/local/lib/python3.5/dist-packages/compose/config/config.py", line 962, in merge_ports
    md[field] = [item for item in sorted(merged.values())]
TypeError: unorderable types: str() < NoneType()
Makefile:57: recipe for target 'compose-build' failed
make: *** [compose-build] Error 1
```

--- a/compose/config/config.py
+++ b/compose/config/config.py
@@ -959,7 +959,7 @@ def merge_ports(md, base, override):
 
     merged = parse_sequence_func(md.base.get(field, []))
     merged.update(parse_sequence_func(md.override.get(field, [])))
-    md[field] = [item for item in sorted(merged.values())]
+    md[field] = [item for item in sorted(merged.values(), key=lambda x: x.target)]
 
 
 def merge_build(output, base, override):

--- a/tests/unit/config/config_test.py
+++ b/tests/unit/config/config_test.py
@@ -1615,6 +1615,22 @@ class ConfigTest(unittest.TestCase):
             'ports': types.ServicePort.parse('5432')
         }
 
+    def test_merge_service_dicts_ports_sorting(self):
+        base = {
+            'ports': [5432]
+        }
+        override = {
+            'image': 'alpine:edge',
+            'ports': ['5432/udp']
+        }
+        actual = config.merge_service_dicts_from_files(
+            base,
+            override,
+            DEFAULT_VERSION)
+        assert len(actual['ports']) == 2
+        assert types.ServicePort.parse('5432')[0] in actual['ports']
+        assert types.ServicePort.parse('5432/udp')[0] in actual['ports']
+
     def test_merge_service_dicts_heterogeneous_volumes(self):
         base = {
             'volumes': ['/a:/b', '/x:/z'],

